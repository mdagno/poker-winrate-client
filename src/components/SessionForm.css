
form{
  color: white;
  width: 50%;
  margin-left: auto;
  margin-right: auto;
}

.form-row{
  padding: 5px;
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-content: center;
}

.form-row > label {
  padding: .5em 1em .5em 0;
  flex: 1;
}
.form-row > input {
  flex: 2;
}

.form-row > textarea {
  flex: 2;
}