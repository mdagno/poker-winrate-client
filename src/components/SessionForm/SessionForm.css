.addSession {
  margin-top: 20px;
}

.addSession form{
  width: 90%;
  color: #222;
  margin-left: auto;
  margin-right: auto;
  font-size: 0.8em;
}

.addSession h2{
	margin-top: 30px;
  font-size: 1.5em;
	text-align: center;
	color: white;
}

fieldset {
  border-radius: 5px;
  padding: 1.25em;
	border: 1px outset white;
  margin-top: 0.5em;
  background-color: #ccc;
}

.form-row{
  padding: 1.25em;
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-content: center;
  text-align: right;
}

.form-row label {
	border: 3px inset white;
	border-radius: 5px;
	padding: 0.2em;
}

.form-row input {
	flex: 1;
}

.form-row textarea {
  flex: 1;
  height: 100px;
  width: 200px;
}

.gameTypeTwo {
	margin: em;
}

.gameTypeTwo label {
	border: none;
	color: white;
}

button {
  box-shadow:inset 0px 1px 3px 0px #91b8b3;
	background:linear-gradient(to bottom, #768d87 5%, #6c7c7c 100%);
	background-color:#131d38;
	border-radius:5px;
	border:1px solid #566963;
	display:inline-block;
	cursor:pointer;
	color:#ffffff;
	font-size: 1em;
	font-weight:bold;
  padding: 10px;
  margin: 2px;
	text-decoration:none;
	text-shadow:0px -1px 0px #2b665e;
	width: 85px;
	text-align: center;
}

.addSession select {
  margin-left: auto;
  margin-right: auto;
	font-size: 1.08em;
	font-family: sans-serif;
	font-weight: 500;
	color: #444;
  line-height: 1.2;
	padding: .6em 1.4em .5em .8em;
	width: 50%;
	max-width: 50%;
	border: 1px solid #aaa;
	box-shadow: 0 1px 0 1px rgba(0,0,0,.04);
	border-radius: .5em;
	-moz-appearance: none;
	-webkit-appearance: none;
	appearance: none;
	background-color: #fff;
	background-image: url('data:image/svg+xml;charset=US-ASCII,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%22292.4%22%20height%3D%22292.4%22%3E%3Cpath%20fill%3D%22%23007CB2%22%20d%3D%22M287%2069.4a17.6%2017.6%200%200%200-13-5.4H18.4c-5%200-9.3%201.8-12.9%205.4A17.6%2017.6%200%200%200%200%2082.2c0%205%201.8%209.3%205.4%2012.9l128%20127.9c3.6%203.6%207.8%205.4%2012.8%205.4s9.2-1.8%2012.8-5.4L287%2095c3.5-3.5%205.4-7.8%205.4-12.8%200-5-1.9-9.2-5.5-12.8z%22%2F%3E%3C%2Fsvg%3E'),
	  linear-gradient(to bottom, #ffffff 0%,#e5e5e5 100%);
	background-repeat: no-repeat, repeat;
	background-position: right .7em top 50%, 0 0;
	background-size: .65em auto, 100%;
}



@media only screen and (min-device-width : 768px) and (max-device-width : 1024px){
	.addSession form{
		font-size: 1em;
	}
}

/* Desktops and laptops ----------- */
@media only screen  and (min-width : 1224px) {
	.addSession form{
		font-size: 1em;
		width: 40%;
	}

}