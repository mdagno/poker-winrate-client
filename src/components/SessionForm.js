import React from 'react'
import './SessionForm.css'

export default class SessionForm extends React.Component {
  render(){
    return(
      <form>
        <legend>
          <h2>Enter Session Details</h2>
        </legend>
        <fieldset>
          <div className='form-row'>
            <label for='live-or-online'>Live or Online: </label>
            <select>
              <option value='Live'>Live game</option>
              <option value='Online'>Online game</option>
            </select>
          </div>
          <div className='form-row'>
          <label for='cash-or-tourney'>Cash or Tournament: </label>
            <select>
              <option value='Cash'>Cash Game</option>
              <option value='Tournament'>Tournament</option>
            </select>
          </div>
          <div className='form-row'>
            <label for='bigBlind'>Big Blind: </label>
            <input className='form-bigBlind' type='text'></input>
          </div>
          <div className='form-row'>
            <label for='buyIn'>Buy-in: </label>
            <input className='form-buyin' type='number'></input>
          </div>
          <div className='form-row'>
            <label for='cashedOut'>Cashed out: </label>
            <input className='form-cashedOut' type='number'></input>
          </div>
          <div className='form-row'>
            <label for='notes'>Notes: </label>
            <textarea type='text' className='form-notes'></textarea>
          </div>
          <div className='form-row'> 
            <button>Submit</button>
            <button>Cancel</button>
          </div>
        </fieldset>
      </form>
    )
  }
}