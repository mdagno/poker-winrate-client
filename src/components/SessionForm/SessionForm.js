import React from 'react'
import './SessionForm.css'
import ApiContext from '../../ApiContext';
import ApiService from '../../services/api-service'

export default class SessionForm extends React.Component {
  static contextType = ApiContext;
  constructor(props){
    super(props)
    this.state = {
        game_type_one: 'Live',
        game_type_two: 'Cash',
        small_blind: 0,
        big_blind: 0,
        buy_in: null,
        cashed_out: null,
        session_length: null,
        notes: '',
      }
  }

  renderSessionForm = () => {
    let sessionForm = this.state.game_type_two
    let renderedForm = '';
    if(sessionForm === 'Cash'){
      renderedForm = (
      <fieldset>
        <div className='form-row'>
          <label htmlFor='gameTypeOne'>Live or Online</label>
          <select id='gameTypeOne' defaultValue ='Live' onChange={e => this.gameTypeSelect('game_type_one', e.target.value)}>
            <option value='Live'>Live game</option>
            <option value='Online'>Online game</option>
          </select>
        </div>
        <div className='form-row'>
          <label htmlFor='smallBlind'>Small Blind</label>
          <input className='form-smallBlind' type='number' id='smallBlind' min='0.01' step='0.01' onChange={e => this.updateField('small_blind', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='bigBlind'>Big Blind</label>
          <input className='form-bigBlind' type='number' id='bigBlind' min='0.01' step='0.01' onChange={e => this.updateField('big_blind', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='form-buyIn'>Buy-in</label>
          <input className='form-buyIn' type='number' step='0.01' onChange={e => this.updateField('buy_in', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='cashedOut'>Cashed out</label>
          <input className='form-cashedOut' type='number' id='cashedOut' step='0.01' onChange={e => this.updateField('cashed_out', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='cashedOut'>Hours played</label>
          <input className='form-SessionLength' type='number' id='sessionLength' step='0.1' onChange={e => this.updateField('session_length', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='notes'>Notes</label>
          <textarea type='text' className='form-notes' id='notes' onChange={e => this.updateField('notes', e.target.value)}></textarea>
        </div>
        <div className='form-row'> 
          <button type='submit'>Submit</button>
          <button type='reset'>Reset</button>
        </div>
      </fieldset>
      )}

      else if(sessionForm === 'Tournament'){
        renderedForm = (
        <fieldset>
          <div className='form-row'>
          <label htmlFor='gameTypeOne'>Live or Online: </label>
          <select id='gameTypeOne' defaultValue ='Live' onChange={e => this.gameTypeSelect('game_type_one', e.target.value)}>
            <option value='Live'>Live game</option>
            <option value='Online'>Online game</option>
          </select>
        </div>
        <div className='form-row'>
          <label htmlFor='form-buyIn'>Buy-in: </label>
          <input className='form-buyin' type='number' step='0.01' onChange={e => this.updateField('buy_in', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='form-cashedOut'>Cashed out: </label>
          <input className='form-cashedOut' type='number' id='cashedOut' step='0.01' onChange={e => this.updateField('cashed_out', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='form-sessionLength'>Hours played: </label>
          <input className='form-sessionLength' type='number' id='sessionLength' step='0.1' onChange={e => this.updateField('session_length', e.target.value)}></input>
        </div>
        <div className='form-row'>
          <label htmlFor='notes'>Notes: </label>
          <textarea type='text' className='form-notes' id='notes' onChange={e => this.updateField('notes', e.target.value)}></textarea>
        </div>
        <div className='form-row'> 
          <button type='submit'>Submit</button>
          <button type='reset'>Reset</button>
        </div>
      </fieldset>
      )
    }
      return renderedForm;
  }
 
  handleSessionSubmit = e => {
    e.preventDefault();
    let newSession = JSON.stringify(this.state)
    ApiService.addSession(newSession)
    .then(res => {
      this.context.addSessionContext(res)
      this.props.history.push('/')
    }) 
  }

  updateField = (field, value) => {
    this.setState({
      [field]: value,
    })
  }

  gameTypeSelect = (gameType, value) => {
    this.setState({
      [gameType]: value,
      small_blind: 0,
      big_blind: 0
    })
  }

  render(){
    return(
      <div className='addSession'>
      <form onSubmit={this.handleSessionSubmit}>
        <legend>
          <h2>Enter Session Details</h2>
        </legend>
        <div className='form-row gameTypeTwo'>
         <label htmlFor='gameTypeTwo'>Game Type: </label>
          <select id='gameTypeTwo' defaultValue='Cash' onChange={e => this.gameTypeSelect('game_type_two', e.target.value)}>
            <option value='Cash'>Cash Game</option>
            <option value='Tournament'>Tournament</option>
          </select>
        </div>
        {this.renderSessionForm()}
      </form>
      </div>
    )
  }
}