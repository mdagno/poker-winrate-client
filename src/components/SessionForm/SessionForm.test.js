import React from 'react';
import SessionForm from './SessionForm';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

describe('SessionForm component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SessionForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('SessionForm renders correctly', () => {
    const tree = renderer
      .create(<SessionForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})

