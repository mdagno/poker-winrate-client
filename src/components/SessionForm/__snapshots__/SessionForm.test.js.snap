// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionForm component SessionForm renders correctly 1`] = `
<div
  className="addSession"
>
  <form
    onSubmit={[Function]}
  >
    <legend>
      <h2>
        Enter Session Details
      </h2>
    </legend>
    <div
      className="form-row gameTypeTwo"
    >
      <label
        htmlFor="gameTypeTwo"
      >
        Game Type: 
      </label>
      <select
        defaultValue="Cash"
        id="gameTypeTwo"
        onChange={[Function]}
      >
        <option
          value="Cash"
        >
          Cash Game
        </option>
        <option
          value="Tournament"
        >
          Tournament
        </option>
      </select>
    </div>
    <fieldset>
      <div
        className="form-row"
      >
        <label
          htmlFor="gameTypeOne"
        >
          Live or Online
        </label>
        <select
          defaultValue="Live"
          id="gameTypeOne"
          onChange={[Function]}
        >
          <option
            value="Live"
          >
            Live game
          </option>
          <option
            value="Online"
          >
            Online game
          </option>
        </select>
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="smallBlind"
        >
          Small Blind
        </label>
        <input
          className="form-smallBlind"
          id="smallBlind"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="bigBlind"
        >
          Big Blind
        </label>
        <input
          className="form-bigBlind"
          id="bigBlind"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="buyIn"
        >
          Buy-in
        </label>
        <input
          className="form-buyin"
          id="buyIn"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="cashedOut"
        >
          Cashed out
        </label>
        <input
          className="form-cashedOut"
          id="cashedOut"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="cashedOut"
        >
          Hours played
        </label>
        <input
          className="form-SessionLength"
          id="sessionLength"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-row"
      >
        <label
          htmlFor="notes"
        >
          Notes
        </label>
        <textarea
          className="form-notes"
          id="notes"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-row"
      >
        <button
          type="submit"
        >
          Submit
        </button>
        <button
          type="reset"
        >
          Reset
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;
